### ISSUE TITLE
Missing `Base.deepcopy_internal` for Group interface

### ISSUE THREAD

**Issue by thofma** [oscar-system/Oscar.jl#860]
Was doing some computations and encountered:
```julia
ERROR: Missing method from Group interface: `Base.deepcopy_internal(::PcGroupElem, ::IdDict)`
Stacktrace:
[...]
```

**Comment by fingolfin** [comment]
Related: https://github.com/oscar-system/GAP.jl/issues/197

**Comment by ThomasBreuer** [comment]
How to reproduce the problem:
``deepcopy([one(small_group(2, 1))])``

**Comment by fingolfin** [comment]
Full backtrace:
```
julia> deepcopy([one(small_group(2, 1))])
ERROR: Missing method from Group interface: `Base.deepcopy_internal(::PcGroupElem, ::IdDict)`
Stacktrace:
 [1] deepcopy_internal(g::PcGroupElem, stackdict::IdDict{Any, Any})
   @ GroupsCore ~/.julia/packages/GroupsCore/63Vm8/src/group_elements.jl:30
 [2] _deepcopy_array_t(x::Array, T::Type, stackdict::IdDict{Any, Any})
   @ Base ./deepcopy.jl:105
 [3] deepcopy_internal
   @ ./deepcopy.jl:92 [inlined]
 [4] deepcopy(x::Vector{PcGroupElem})
   @ Base ./deepcopy.jl:26
 [5] top-level scope
   @ REPL[2]:1
```

So GroupsCore forces us to be nice and implement the interface properly :-). OK, so we just have to do that. @ThomasBreuer will work on it.

**Comment by fingolfin** [comment]
Resolved by #1037
